### PR TITLE
[Feat] Adds compression functionalities for WeiGit

### PR DESCRIPTION
**Pull request by riohib**
1. Adds functions for sparsification of a state_dict and creating a mask for a state_dict with a given  sparsity level.
2. Functions for weight thresholding in a state_dict
3. FFT and IFFT of a state_dict and options of thresholding after Fourier transform, creating Signal Sparse Tensor (SST)
4. Reconstruction function from SST and Delta Sparse Tensor (DST)

## What does this PR do?
Fixes # (issue).

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/main/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/main/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: fairscale/experimental/wgit/compression.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
+from collections import OrderedDict
```
Comment (min-xu-ai) [inline fairscale/experimental/wgit/compression.py:1]: don't forget the copyright header too. I will take a look when this is ready. Add simple tests with fixed tensors and expected golden values to ensure the coverage for the new code here. Thanks!

File: fairscale/experimental/wgit/compression.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        dim (int)
+            The target dimension for which
+
+    """
+    thresholds_l = torch.quantile(w_tensor, q=sparsity, dim=1)  # type: ignore
+    linear_mask = w_tensor > thresholds_l.unsqueeze(1)
+    return linear_mask
+
+
+def compressible_layers(model: nn.Module) -> List[str]:
```
Comment (riohib) [inline fairscale/experimental/wgit/compression.py:175]: @min-xu-ai this function currently seems to be quite ad-hoc. May be we can delegate the task of finding the list of layers to be compressed to the callers of other functions, and take an argument with the list?
